modules/openapi-generator/src/main/resources/cpp-restbed-server/model-header.mustache

```diff
@@ -21,30 +22,39 @@ namespace {{this}} {
 /// <summary>
 /// {{description}}
 /// </summary>
-class {{declspec}} {{classname}}
+{{#circularReferences}}
+class  {{{this}}};
+{{/circularReferences}}
+class {{declspec}} {{classname}} {{#interfaces}}{{#-first}}:{{/-first}}{{^-first}},{{/-first}} public {{{this}}}{{/interfaces}}
 {
 public:
     {{classname}}();
     virtual ~{{classname}}();
     
-    std::string toJsonString();
+    std::string toJsonString(bool prettyJson = false);
     void fromJsonString(std::string const& jsonString);
+    boost::property_tree::ptree toPropertyTree();
+    void fromPropertyTree(boost::property_tree::ptree const& pt);
 
     /////////////////////////////////////////////
     /// {{classname}} members
-    
     {{#vars}}
+
     /// <summary>
     /// {{description}}
     /// </summary>
     {{{dataType}}} {{getter}}() const;
     void {{setter}}({{{dataType}}} value);
     {{/vars}}
-
 protected:
     {{#vars}}
     {{{dataType}}} m_{{name}};
     {{/vars}}
+    {{#vars}}
+    {{#isEnum}}
+    std::vector<{{{dataType}}}> m_{{enumName}};
```

Comment:
Yup, makes sense.

In hindsight I could have created a member to save the value that is an actual ENUM instead, instead of keeping it as a `std::string`, but that seemed just a tad easier.